CHANGELOG.md

```diff
@@ -37,6 +37,8 @@ _ `_optional` subpackage for managing optional dependencies
 ### Deprecations
 - `SequentialGreedyRecommender` class replaced with `BotorchRecommender`
 
+### Deprecated
```

Comment:
Just a quick question: Since this PR mentions another PR and some "open problems", I just wanted to ask if this PR is already ready for review since the PR that is mentioned is closed by now.

Reply:
@AdrianSosic question +1

Reply:
All good, the other PR has been merged, so this one here should be ready to go. Have also just rebased it on latest main 👍🏼 It's mostly a maintenance PR that moves us a bit closer to fixing some of the remaining open issues (e.g. how callables are provided for the simulation etc). So just some cleanup work upfront